docs/advanced/advanced_troubleshooting.md

```diff
@@ -0,0 +1,170 @@
+# Troubleshooting
+Use this section for advanced troubleshooting information for your CSI driver.
+
+- [Verifying the CSI driver is running](#Verifying-the-CSI-driver-is-running)
+- [Multipath troubleshooting](#Multipath-troubleshooting)
+- [General troubleshooting](#General-troubleshooting)
+
+## Verifying the CSI driver is running
+
+  Verify that the CSI driver is running. (Make sure the csi-controller pod status is _Running_).
+
+    $> kubectl get all -n <namespace> -l csi
+    NAME                                        READY STATUS  RESTARTS  AGE
+    pod/ibm-block-csi-controller-0              6/6   Running 0         2h
```

Comment:
7/7

Reply:
@lisyukanna  is it sufficient to just write the command without the output here?

Reply:
sure

Reply:
@roysahar ? @barakda ? Let me know what to do here. I don't remember exactly where I got it but probably from QA at some point.

Reply:
Here is a better status but I'm agree with @oriyarde, we will need to maintain it for ever.. 

[root@stk8s001 brkd]# kubectl get all -n default -l csi
NAME                                         READY   STATUS    RESTARTS   AGE
pod/ibm-block-csi-controller-0               7/7     Running   0          21m
pod/ibm-block-csi-node-57nm2                 3/3     Running   0          21m
pod/ibm-block-csi-node-ssjq4                 3/3     Running   0          21m
pod/ibm-block-csi-operator-9f6f49c77-88fft   1/1     Running   0          22m

Reply:
@barakda  I think that the question here is regarding:

> ## Verifying the CSI driver is running
> 
>   Verify that the CSI driver is running. (Make sure the csi-controller pod status is _Running_), using the `$> kubectl get all -n <namespace> -l csi` command.

Is there a reason why we're specifically mentioning csi-controller pod status and not just "all components are running"?

Reply:
@rivkap , I don't know why we have mentioned here csi-controller and not all components, I guess it is just a mistake, pls change

Reply:
DONE

Reply:
also, why do we use `all` and not just `pod`?
if we want to keep `all`, we cannot just use "running", because the non-pod objects should be "ready"

Reply:
this is why specifics during commenting/reviews are important :)

So, what should it be now?
Is this correct?

`Verify that the CSI driver is running. (Make sure that all pods are in the  _Running_ state), using the `$> kubectl get all -n <namespace> -l csi` command.
`

Reply:
not entirely accurate, since `get all` shows more than pods.
if we decide to limit the scope to pods-only, then the command itself should have `pod`/`pods` instead of `all`

Reply:
So the options are as follows??

> Verify that the CSI driver is running. Make sure that all pods are in the _Running_ state, using the `$> kubectl get pods -n -l csi` command.

or

> Verify that the CSI driver is in the _Ready_ state, using the `$> kubectl get all-n -l csi` command.

Reply:
the fist option is correct, the second is not, because `get all` also includes pods, that should be running (it's a mixed output)

Reply:
also I think @roysahar once pointed out to me that even the first option is not completely accurate, since the `csi` label (used in `-l csi`) is not necessarily unique to our product (if e.g. there are other `csi` driver installed)

Reply:
I wonder if he found a better label

Reply:
@roysahar  @barakda  What command (and output status) do you want me to put in for "Verifying the CSI driver is running"?
Please give me full specifics (full command and full output that you want documented). thanks

Reply:
@rivkap,
I would suggest to use the basic known command to get all driver pods:
> kubectl get pods -l csi -n <namespace>

Reply:
we need the csi label as the customer could have more pods on the list..

Reply:
Per @barakda 's comment above, I am leaving the content of this section as:

> Verify that the CSI driver is running. (Make sure that all components are in the  _Running_ state), using the `$> kubectl get all -n <namespace> -l csi` command.

Reply:
DONE
@oriyarde

Reply:
I'm not sure I follow Barak's reasoning, but ok

Reply:
resolving on behalf of Arbel